python/rmm/_lib/device_buffer.pyx

```diff
@@ -46,6 +46,7 @@ cdef class DeviceBuffer:
                   uintptr_t stream=0):
```

Comment:
@jakirkham we should add documentation to the init function here to clarify the synchronization behavior with regards to streams here.

Reply:
Agreed. Thanks for mentioning. Have added docs for the constructor along with this note.

Reply:
Also double checked that we have this note in other places where we synchronize.